dicom_utils/cli/dicom_types.py

```diff
@@ -37,25 +37,14 @@ def check_file(path: Path, args: argparse.Namespace) -> Optional[Tuple[Path, int
             image_type = image_type.value
             if isinstance(image_type, str):
                 image_type = [image_type]
-            image_type_dict = process_image_type(dcm)
-            simple_image_type = get_simple_image_type(image_type_dict)
-            num_frames = dcm.get("NumberOfFrames", 1)
+            img_type = ImageType.from_dicom(dcm)
+            simple_image_type = img_type.to_simple_image_type()
+            num_frames = img_type.NumberOfFrames or 1
```

Comment:
If NumberOfFrames can be `int` instead of `Optional[int]` (see comment below), then we can drop the `or 1`.